daemon/user.go

```diff
@@ -84,3 +91,98 @@ func (d *daemon) InviteUser(ctx context.Context, req *pb.InviteUserRequest) (*pb
 		Key: k.String(),
 	}, nil
 }
+
+// ListUsers function lists users which are eligible to access commandline and
+// webclient of Haaukins, who have credeentials will be displayed
+func (d *daemon) ListUsers(ctx context.Context, req *pb.Empty) (*pb.ListUsersResponse, error) {
+	var usersResp []*pb.ListUsersResponse_UserInfo
+	requester, err := getUserFromIncomingContext(ctx)
+	if err != nil {
+		log.Warn().Msgf("User credentials not found ! %v  ", err)
+		return &pb.ListUsersResponse{}, err
+	}
+
+	if !requester.SuperUser {
+		return &pb.ListUsersResponse{}, NoPrivilegeToList
+	}
+
+	//todo: add users' events as well.
+
+	for _, usr := range d.users.ListUsers() {
+		usersResp = append(usersResp, &pb.ListUsersResponse_UserInfo{
+			Username:    usr.Username,
+			Name:        usr.Name,
+			Surname:     usr.Username,
+			Email:       usr.Email,
+			CreatedAt:   usr.CreatedAt.Format(displayTimeFormat),
+			IsSuperUser: usr.SuperUser,
+		})
+	}
+
+	return &pb.ListUsersResponse{Users: usersResp}, nil
+
+}
+
+//DestroyUser function deletes user only admin accounts
+// An admin account should not delete another admin account
+func (d *daemon) DestroyUser(ctx context.Context, request *pb.DestroyUserRequest) (*pb.DestroyUserResponse, error) {
+
+	requester, err := getUserFromIncomingContext(ctx)
+	if err != nil {
+		log.Warn().Msgf("User credentials not found ! %v  ", err)
+		return &pb.DestroyUserResponse{}, err
+	}
+
+	if !requester.SuperUser {
+		return &pb.DestroyUserResponse{}, NoPrivilegeToDelete
```

Comment:
Modify the response in case of error according the proto file changed (for each error response)